service/s3/api_client.go

```diff
@@ -452,3 +452,125 @@ func addRequestResponseLogging(stack *middleware.Stack, o Options) error {
 		LogResponseWithBody: o.ClientLogMode.IsResponseWithBody(),
 	}, middleware.After)
 }
+
+// PresignOptions represents the presign client options
+type PresignOptions struct {
+	// ClientOptions are list of functional options to mutate client options used by
+	// presign client
+	ClientOptions []func(*Options)
+	// Presigner is the presigner used by the presign url client
+	Presigner v4.HTTPPresigner
```

Comment:
Should this be an interface similar to the Signer for the API client?

```
	// Signature Version 4 (SigV4) Signer
	HTTPSignerV4 HTTPSignerV4
```

Reply:
`v4.HTTPPresigner` is an interface defined for v4 signer.

Reply:
Yeah i think the intention here is to have the client define its own interface similar to the way it does for `HTTPSignerV4`